perf/blogbench.py

```diff
@@ -0,0 +1,62 @@
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import archive
+from avocado.utils.software_manager import SoftwareManager
+from avocado.core import data_dir
+
+
+class Blogbench(Test):
+
+    ''' Blogbench will start the required threads and the test will run
+        according to the args given.A final "score" will then be given as
+        an indication of read and write performance '''
+
+    def setUp(self):
+        sm = SoftwareManager()
+        # Check for basic utilities
+        deps = ['gcc', 'make']
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        url = 'https://download.pureftpd.org/blogbench/blogbench-1.1.tar.bz2'
+        blogbench_url = self.params.get('blogbench_url',
+                                        default=url)
+        blogbench_tarball = self.fetch_asset(blogbench_url, expire='7d')
+        archive.extract(blogbench_tarball, self.srcdir)
+        blogbench_version = os.path.basename(blogbench_tarball
+                                             .split('.tar.')[0])
+        self.blogbench_dir = os.path.join(self.srcdir, blogbench_version)
+        os.chdir(self.blogbench_dir)
+        patch = self.params.get('patch', default='config_guess.patch')
+        process.run('patch -p1 config.guess %s' % self.datadir +
+                    '/' + patch, shell=True)
```

Comment:
please use  os.path.join

code can be 
`process.run('patch -p1 config.guess %s' % os.path.join(self.datadir,patch),shell=True)`